src/Fixer/Phpdoc/PhpdocOrderFixer.php

```diff
@@ -118,8 +118,12 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
         $configurationOrder = [];
         foreach ($this->configuration['order'] as $type) {
             $configurationOrder[] = $type;
-            $configurationOrder[] = 'phpstan-'.$type;
-            $configurationOrder[] = 'psalm-'.$type;
+            if (!\in_array('phpstan-'.$type, $this->configuration['order'], true)) {
+                $configurationOrder[] = 'phpstan-'.$type;
```

Comment:
The fallback should be probably added after anything else.

Ex. when order is specified like `param, phpstan-param`, with the current PR, `param, psalm-param, phpstan-param` order will be kept, but I would expect `param, phpstan-param, psalm-param`.

Reply:
Hmm, that's a reasonable point. Although I think there are potentially other cases where this behaviour can produce slightly-unexpected results especially if you specify order for some variants but not others.

Attempting to check & find the right insertion point for the prefixed variants would add even more overhead to this fixer at runtime - I'm not sure if that's desirable?

IMO the priority is to fix the bug in the current implementation (in particular that it can propose completely deleting annotations). 

My gut feel is that it may be appropriate to say that if someone is using param, phpstan-param and psalm-param in the same project then they should either use the default ordering or specify all three in the order they want them.